typhon/cloudmask/aster.py

```diff
@@ -531,29 +529,23 @@ def get_latlon_grid(self, channel="1"):
             ]
         )
 
-        # Image corner coordinates (file, NOT swath) according to the ASTER
-        # Users Handbook p.57.
-        LL = self._convert_metastr(
-            self.meta["LOWERLEFT"], dtype=tuple
-        )  # (latitude, longitude)
-        LR = self._convert_metastr(self.meta["LOWERRIGHT"], dtype=tuple)
-        UL = self._convert_metastr(self.meta["UPPERLEFT"], dtype=tuple)
-        UR = self._convert_metastr(self.meta["UPPERRIGHT"], dtype=tuple)
-        lat = np.array([[UL[0], UR[0]], [LL[0], LR[0]]])
-        lon = np.array([[UL[1], UR[1]], [LL[1], LR[1]]])
+        lat, lon = self.read_coordinates(channel=channel)
+
+        latidx = np.arange(11) * imagedimension.lat / 10
+        lonidx = np.arange(11) * imagedimension.lon / 10
```

Comment:
Are the image dimensions always dividable by ten? Otherwise you could use an integer division `x // 10` to ensure an `int` result

Reply:
The ASTER L1B user guide specifies geolocation boxes, their sizes, and the corresponding lattice points (which don't change).`latidx` and `lonidx` are those lattice points and are only needed for defining the interpolation function and not necessarily need to be `int`.